doc/design/proposal-ambient-loaders.md

```diff
@@ -0,0 +1,45 @@
+# Ambient Loaders Design
+
+## Problem
+
+Loaders let side logic be applied when resolving import statements. However, this only applies to the main application: loaders themselves aren’t currently affected by any other loaders. As a result, 3rd-party loaders cannot be injected if accessing them requires going through another loader. For instance, the following doesn’t work:
+
+```
+node --loader pnp --loader ts-node ./my-tool.mts
```

Comment:
When you say that it doesn’t work, I assume what you mean is that it doesn’t work when there’s no `node_modules` folder. In other words, if this were a more traditional app with a `package.json` that contained `ts-node` as a dependency, and someone had run `npm install`, it _would_ work; it only doesn’t work in this case because you’re trying to avoid the install step and use PNP instead. Am I understanding you correctly?

Reply:
PnP is just an example. A loader that would allow node_modules folders to be compressed archives would have similar problems.

Reply:
Sure. What I meant was, you should explain this in the text, that you're assuming no `node_modules` here.

Reply:
Clarified 👍